### PR TITLE
Use HolidayBase::_add_holiday_* family methods [b]

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

This PR continues #1368. Update countries starting w/ b.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/burundi.py [hunk 1]
```diff
@@ -38,13 +37,15 @@ def __init__(self, *args, **kwargs):
...
-            )
-        return dt
+    def _add_holiday(self, name: str, *args) -> Optional[date]:
+        dt = args if len(args) > 1 else args[0]
+        dt = dt if isinstance(dt, date) else date(self._year, *dt)
+
+        dt_added = super()._add_holiday(name, dt)
+        if self.observed and dt_added and self._is_sunday(dt_added):
+            dt_added = super()._add_holiday("%s (Observed)" % self[dt], dt + td(days=+1))
+
```
Comment (KJhellico) [inline holidays/countries/burundi.py:47]:
```suggestion
        dt_added = super()._add_holiday(name, *args)
        if self.observed and dt_added and self._is_sunday(dt_added):
            dt_added = super()._add_holiday("%s (Observed)" % name, dt_added + td(days=+1))
```

Comment (arkid15r) [reply holidays/countries/burundi.py:47]: Yeah, those are for a future refactoring of `HolidayBase::_add_holiday` as it's going to stop accepting *args and will take name/dt instead. Not sure if it's better to keep it now or remove.

Comment (KJhellico) [reply holidays/countries/burundi.py:47]: Then let's keep it now.

**Review by KJhellico — COMMENTED** [review]
LGTM.

**Pull request merged**